android/app/src/main/kotlin/com/jabook/app/jabook/audio/PlayerErrorHandler.kt

```
@@ -0,0 +1,220 @@
+                if (player.isPlaying || player.playWhenReady) player.playWhenReady = true
+            } catch (e: Exception) {
+                LogUtils.e(TAG, "Failed to skip to next track", e)
+            }
+        } else {
+            LogUtils.w(TAG, "No more tracks, pausing")
+            try { player.playWhenReady = false } catch (_: Exception) {}
+        }
+    }
+
+    /** Skips to next available track when current is unavailable (file check). */
+    fun skipToNextAvailableTrack(currentIndex: Int, previousIndex: Int) {
```

detekt (1.23.8)

[warning] The function skipToNextAvailableTrack appears to be too complex based on Cognitive Complexity (complexity: 27). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)